File: Selenium/Activities/activity13.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;

import java.util.List;

public class activity13 {
    public static void main(String args[])
    {
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver = new FirefoxDriver();
        driver.get("https://v1.training-support.net/selenium/tables");
        System.out.println("Title of the page is: "+driver.getTitle());
        List<WebElement> rows = driver.findElements(By.xpath("//table[contains(@class,' striped')]/tbody/tr"));
        System.out.println("Rows: "+rows.size());
        List<WebElement> cols=driver.findElements(By.xpath("//table[contains(@class,' striped')]/tbody/tr[1]/td"));
        System.out.println("Columns: "+cols.size());
        List<WebElement> thirdRow=driver.findElements(By.xpath("//table[contains(@class,' striped')]/tbody/tr[3]/td"));
        System.out.println("Third row cell values: ");
        for(WebElement cell:thirdRow) {
            System.out.println(cell.getText());
        }
       WebElement cellValue = driver.findElement(By.xpath("//table[contains(@class,' striped')]/tbody/tr[2]/td[2]"));
        System.out.println("Second row, second cell value is: "+cellValue.getText());
        driver.close();
    }
}

File: Selenium/Activities/activity13.py
from selenium.webdriver.common.by import By
from selenium import webdriver
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager
driver = webdriver.Firefox(service=FirefoxService(GeckoDriverManager().install()))
driver.get("https://v1.training-support.net/selenium/tables")
print("Page title is: ", driver.title)
columns = driver.find_elements(By.XPATH, "//table[contains(@class, 'striped')]/tbody/tr[1]/td")
print("Number of columns: ", len(columns))
rows = driver.find_elements(By.XPATH, "//table[contains(@class, 'striped')]/tbody/tr")
print("Number of rows: ", len(rows))
third_row = driver.find_elements(By.XPATH, "//table[contains(@class, 'striped')]/tbody/tr[3]/td")
print("Cell values of third row:")
for row in third_row:
    print("Cell value: ", row.text)
second_row_second_cell = driver.find_element(By.XPATH, "//table[contains(@class, 'striped')]/tbody/tr[2]/td[2]")
print("Cell value at second row and second column: ", second_row_second_cell.text)
driver.quit()
